### ISSUE TITLE
Cannot use object of type WP_Error as array

### ISSUE THREAD

**Issue by virik** [krokedil/klarna-checkout-for-woocommerce#116]
Our WC fatal error log says:

2018-10-21T12:04:21+00:00 CRITICAL Uncaught Error: Cannot use object of type WP_Error as array in [domain]/wp-content/plugins/klarna-checkout-for-woocommerce/includes/class-klarna-checkout-for-woocommerce-api.php:193
Stack trace:
#0 [domain]/wp-content/plugins/klarna-checkout-for-woocommerce/includes/class-klarna-checkout-for-woocommerce-checkout-form-fields.php(25): Klarna_Checkout_For_WooCommerce_API->request_pre_get_order('f1e67f5c-ed8b-4...')
#1 [domain]/wp-content/plugins/klarna-checkout-for-woocommerce/includes/class-klarna-checkout-for-woocommerce-checkout-form-fields.php(31): Klarna_Checkout_For_Woocommerce_Checkout_Form_Fields::maybe_set_klarna_order()
#2 [domain]/wp-content/plugins/klarna-checkout-for-woocommerce/includes/class-klarna-checkout-for-woocommerce-ajax.php(248): Klarna_Checkout_For_Woocommerce_Checkout_Form_Fields::maybe_set_customer_email()

**Comment by virik** [comment]
And this:

2018-10-20T22:29:42+00:00 CRITICAL Uncaught Error: Class 'Klarna_Checkout_For_WooCommerce_API' not found in [domain]/wp-content/plugins/klarna-checkout-for-woocommerce/klarna-checkout-for-woocommerce.php:321
Stack trace:
#0 [domain]/wp-content/plugins/klarna-checkout-for-woocommerce/klarna-checkout-for-woocommerce.php(163): Klarna_Checkout_For_WooCommerce->init_gateways()
#1 [domain]/wp-includes/class-wp-hook.php(286): Klarna_Checkout_For_WooCommerce->init('')
#2 [domain]/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters(NULL, Array)
#3 [domain]/wp-includes/plugin.php(453): WP_Hook->do_action(Array)
#4 [domain]/wp-settings.php(327): do_action('plugins_loaded')
...

**Comment by MichaelBengtsson** [comment]
@virik The wp_error handeling is something im looking into at the moment and should have a fix for soon. However, the fact that you are getting a wp_error is something we need to look into. When we have a better handeling of the error we can see better what it is returning and should be able to give you a better response on what needs fixing.

**Comment by bonny** [comment]
I get the same error sometimes. Below is the response from `wp_safe_remote_get`. 

This is in the Klarna Sandbox environment btw, maybe they have slower API responses or maybe the API is down sometimes.

```
WP_Error Object
(
    [errors] => Array
        (
            [http_request_failed] => Array
                (
                    [0] => cURL error 28: Operation timed out after 5073 milliseconds with 0 bytes received
                )

        )

    [error_data] => Array
        (
        )

)
```

**Comment by NiklasHogefjord** [comment]
Hi,

This should be fixed in 29b2f2fe3d1e15218193e7b9caea1dd1dda59842. It's included in the latest version (https://github.com/krokedil/klarna-checkout-for-woocommerce/releases/tag/1.7.4).

**Comment by Tobias-Conrad** [comment]
got the issue again in Version 1.8.1 

Fatal error: Uncaught Error: Cannot use object of type WP_Error as array in /var/www/clients/client1/web3/web/wp-content/plugins/klarna-checkout-for-woocommerce/includes/class-klarna-checkout-for-woocommerce-api.php:328 Stack trace: #0 /var/www/clients/client1/web3/web/wp-content/plugins/klarna-checkout-for-woocommerce/includes/klarna-checkout-for-woocommerce-functions.php(10): Klarna_Checkout_For_WooCommerce_API->get_order() #1 /var/www/clients/client1/web3/web/wp-content/plugins/klarna-checkout-for-woocommerce/templates/klarna-checkout.php(30): kco_wc_show_snippet() #2 /var/www/clients/client1/web3/web/wp-content/plugins/woocommerce/includes/wc-core-functions.php(211): include('/var/www/client...') #3 /var/www/clients/client1/web3/web/wp-content/plugins/woocommerce/includes/shortcodes/class-wc-shortcode-checkout.php(283): wc_get_template('checkout/form-c...', Array) #4 /var/www/clients/client1/web3/web/wp-content/plugins/woocommerce/includes/shortcodes/class-wc-shortcode-checkout.php(67): WC_Shortcode_Checkout::checkout in /var/www/clients/client1/web3/web/wp-content/plugins/klarna-checkout-for-woocommerce/includes/class-klarna-checkout-for-woocommerce-api.php on line 328

**Comment by Tobias-Conrad** [comment]
issue introduced in version 1.8.0 in 1.7.9 checkout is working.
very bad that i got know about the issue so late and not full filling orders since days!

**Comment by NiklasHogefjord** [comment]
This should be fixed in 77857e2297bd8f6aaca2e470de63d105152aad30. Currently available in Develop branch. We will test and release it later this week.
